federation/pkg/kubefed/init/init.go

```diff
@@ -133,13 +134,18 @@ type entityKeyPairs struct {
 // See the design doc in https://github.com/kubernetes/kubernetes/pull/34484
 // for details.
 func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Command, args []string) error {
+	ips := []string{}
+	hostnames := []string{}
```

Comment:
See my comment below. After making that change, you can avoid this hoisting by letting these variables to be declared+initialized on function return.